ostd/src/task/scheduler/mod.rs

```diff
@@ -114,27 +112,45 @@ pub(crate) fn might_preempt() {
     yield_now();
 }
 
-/// Blocks the current task unless `has_woken` is `true`.
-pub(crate) fn park_current(has_woken: &AtomicBool) {
+/// Blocks the current task unless `has_woken()` returns `true`.
+///
+/// Note that this method may return due to spurious wake events. It's the caller's responsibility
+/// to detect them (if necessary).
+pub(crate) fn park_current<F>(mut has_woken: F)
+where
+    F: FnMut() -> bool,
```

Comment:
Taking a mutable closure `FnMut` looks weird for a closure that is supposed to check if a condition turns true.

Reply:
Oh, that's a good catch! Thanks!